pkg/validate/external.go

```diff
@@ -0,0 +1,72 @@
+// Copyright 2021 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package validate
+
+import (
+	"context"
+	"encoding/json"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+)
+
+// ValidatorEntrypointsEnv should be set to a Unix path list ("/path/to/e1.sh:/path/to/e2")
+// containing the list of entrypoints to external (out of code tree) validator scripts
+// or binaries to run. Requirements for entrypoints:
+// - Entrypoints must be executable by the user running the parent process.
+// - The stdout output of an entrypoint *must* conform to the JSON representation
+//   of Result so results can be parsed and collated with other internal validators.
+// - An entrypoint should exit 1 and print output to stderr only if the entrypoint itself
+//   fails for some reason. If the bundle fails to pass validation, that information
+//   should be encoded in the Result printed to stdout as a Type="error".
+//
+// WARNING: the script or binary at the base of this path will be executed arbitrarily,
+// so make sure you check the contents of that script or binary prior to running.
+const ValidatorEntrypointsEnv = "OPERATOR_SDK_VALIDATOR_ENTRYPOINTS"
+
+// GetExternalValidatorEntrypoints returns a list of external validator entrypoints
+// retrieved from ValidatorEntrypointsEnv and true if set. If not set or set to the empty string,
+// GetExternalValidatorEntrypoints returns false.
+func GetExternalValidatorEntrypoints() ([]string, bool) {
+	entrypoints, isSet := os.LookupEnv(ValidatorEntrypointsEnv)
+	if !isSet || strings.TrimSpace(entrypoints) == "" {
+		return nil, false
+	}
+	return filepath.SplitList(entrypoints), true
+}
```

Comment:
You might want to do some additional sanity checks on these values being provided.

1. Does the path being provided correlate to something on the file system?
1. Is this a directory?
2. Is this really an executable file?

Reply:
If validation of the entry point paths fails, probably makes sense to log the issue and strip that one out of the returned list so the rest of the system won't act on garbage entry points.

Reply:
`exec.Cmd.Run()` will return a fairly indicative error if a file cannot be found by path, is a directory, or if the file is not executable.